todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TaskViewModel.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.architecture.blueprints.todoapp;
+
+import android.content.Context;
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+import android.databinding.ObservableField;
+
+import com.android.annotations.Nullable;
+import com.example.android.architecture.blueprints.todoapp.data.Task;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksDataSource;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
+
+
+/**
+ * TODO
+ */
+public abstract class TaskViewModel extends BaseObservable
+        implements TasksDataSource.GetTaskCallback, SnackBarChangedCallback.SnackBarViewModel {
+
+    public final ObservableField<String> snackBarText = new ObservableField<>();
+
+    private static final String TAG = "TaskViewModel";
+
+    private final TasksRepository mTasksRepository;
+
+    private final Context mContext;
+
+    @Nullable
+    private Task mTask;
+
+    private boolean mIsDataLoading; // TODO observable field?
+
+    public TaskViewModel(Context context, TasksRepository tasksRepository) {
+        mContext = context;
+        mTasksRepository = tasksRepository;
+    }
+
+    public void start(String taskId) {
```

Comment:
The correct diagram is

https://raw.githubusercontent.com/wiki/googlesamples/android-architecture/images/mvvm-databinding.png

There is no presenter in this MVVM variant.